Patch-Source: https://github.com/archlinux/svntogit-packages/blob/afdf00ac125851675c1599ab46545f6c58cfb655/trunk/add-av_stream_get_first_dts-for-chromium.patch
From 95aab0fd83619408995720ce53d7a74790580220 Mon Sep 17 00:00:00 2001
From: "liberato@chromium.org" <liberato@chromium.org>
Date: Wed, 7 Jul 2021 19:01:22 -0700
Subject: [PATCH] Add av_stream_get_first_dts for Chromium

[foutrelis: adjust for new FFStream struct replacing AVStreamInternal]
---
 libavformat/avformat.h | 4 ++++
 libavformat/utils.c    | 7 +++++++
 2 files changed, 11 insertions(+)

diff --git a/libavformat/avformat.h b/libavformat/avformat.h
index 56c1c80..75221d9 100644
--- a/libavformat/avformat.h
+++ b/libavformat/avformat.h
@@ -1202,6 +1202,10 @@ typedef struct AVStreamGroup {

 struct AVCodecParserContext *av_stream_get_parser(const AVStream *s);

+// Chromium: We use the internal field first_dts vvv
+int64_t    av_stream_get_first_dts(const AVStream *st);
+// Chromium: We use the internal field first_dts ^^^
+
 #define AV_PROGRAM_RUNNING 1

 /**
diff --git a/libavformat/demux_utils.c b/libavformat/demux_utils.c
index 86f5512..8b535a8 100644
--- a/libavformat/demux_utils.c
+++ b/libavformat/demux_utils.c
@@ -29,6 +29,13 @@
 #include "demux.h"
 #include "internal.h"

+// Chromium: We use the internal field first_dts vvv
+int64_t av_stream_get_first_dts(const AVStream *st)
+{
+  return cffstream(st)->first_dts;
+}
+// Chromium: We use the internal field first_dts ^^^
+
 struct AVCodecParserContext *av_stream_get_parser(const AVStream *st)
 {
     return cffstream(st)->parser;

